freqtrade/configuration/arguments.py

```diff
@@ -147,6 +153,22 @@ def _build_subcommands(self) -> None:
         list_timeframes_cmd.set_defaults(func=start_list_timeframes)
         self._build_args(optionlist=ARGS_LIST_TIMEFRAMES, parser=list_timeframes_cmd)
 
+        # Add list-markets subcommand
+        list_markets_cmd = subparsers.add_parser(
+            'list-markets',
+            help='Print markets on exchange.'
+        )
+        list_markets_cmd.set_defaults(func=partial(start_list_markets, pairs_only=False))
+        self._build_args(optionlist=ARGS_LIST_PAIRS, parser=list_markets_cmd)
```

Comment:
Based on your description in the docs, using `ARGS_LIST_PAIRS` for `list-markets` might be wrong.

`list-pairs` lists everything with / between (so ETH/BTC, ...).
`list-markets` lists all pairs - but how can it then filter on `--quote` or `--base ETH`? It does not have these 2 parts ... so these 2 flips should be left out - no?

I also don't see the difference in the 2 commands - markets (non-pairs) are not tradeable anyway, so why should we even list them ... ? 
What exchange do i need to use to see the difference? neither for bittrex, nor for binance i get different results.

Reply:
> What exchange do i need to use to see the difference? neither for bittrex, nor for binance i get different results.

Well, the following command:
```
$ for i in `freqtrade list-exchanges -1`; do echo "$i:"; freqtrade list-pairs --exchange $i --all --print-list | grep -o '[[:digit:]]* pairs'; freqtrade list-markets --exchange $i --all --print-list | grep -o '[[:digit:]]* markets'; done
```
gives us:
```
...
coinex:
227 pairs
338 markets
...
kraken:
99 pairs
110 markets
...
okex:
421 pairs
448 markets
okex3:
421 pairs
457 markets
^^ different for okex and okex3!!!
...
```

Also BitMEX, which is in the 'bad' exchanges and I think some others in this list.

Reply:
As for the first part of the comment -- pairs are markets with '/' character. Other markets can be indexes, futures, etc. They also have 'quote' (quote currency, our stake currency for pairs traded by the bot) field properly filled by ccxt. The 'base' field is often strange, though (see `freqtrade list-markets --exchange coinex -a`, for example), but can be ordinal currencies... They also have price, can be interesting as informative pairs (not pairs, informative markets, to be precise). Some can also be traded (futures). In principle, we can also trade them in the future, if we get rid of checking pairs by parsing the symbol and compare the stake_currency with the value of the 'quote' field in the market structure...

When we'll move from pairs to markets (whitilisted/blacklisted markets, informative markets i.o pairs, etc), the list-pair subcommand can be deleted...

Reply:
getting rid of that is a long shot... i've been thinking about this too - but there's 200 steps between now and that point ... 

but this explains it - thanks.